rl4j/rl4j-core/src/main/java/org/deeplearning4j/rl4j/observation/Observation.java

```diff
@@ -25,7 +25,7 @@
  *
  * @author Alexandre Boulanger
  */
-public class Observation {
+public class Observation implements Encodable {
```

Comment:
I really don't like `Encodable` being reintroduced everywhere, especially on new features. I'm still not sure how this helps remove `Encodable` but I'll trust you on this.

Reply:
Encodable and observation are essentially the same thing now. they effectively implement the same interfaces, It means that theya re interchangable and backwardly compatible, if we remove encodable and replace things like Box with Observation then there would now be very few code changes.